src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectAdapterProvider.cs

```diff
@@ -25,7 +26,7 @@ internal class VsProjectAdapterProvider : IVsProjectAdapterProvider
             IVsProjectThreadingService threadingService)
             : this(
                   threadingService,
-                  new AsyncLazy<SVsSolution>(() => serviceProvider.GetServiceAsync<SVsSolution>(), threadingService.JoinableTaskFactory))
+                  new AsyncLazy<SVsSolution>(() => serviceProvider.GetServiceAsync<SVsSolution, SVsSolution>(throwOnFailure: false), threadingService.JoinableTaskFactory))
```

Comment:
When looking at this, is the one that actually hits this. It performed a thread pool blocking cast to SVsSolution.

Reply:
That's great to hear!